docs/specifications/tests/DNSSEC-TP/dnssec05.md

```diff
@@ -1,55 +1,125 @@
+use private algorithms. The IANA registry of [DNSSEC Algorithm Numbers]
+specifies which algorithms to use.
 
-The DNSKEY record is described in section 2 of [RFC 4034](
-https://tools.ietf.org/html/rfc4034#section-2).
+The DNSKEY record is defined in [RFC 4034, section 2].
 
-The IANA registry of DNSKEY algorithm numbers is in the [dns-sec-alg-numbers](
-https://www.iana.org/assignments/dns-sec-alg-numbers/dns-sec-alg-numbers.xml).
+## Inputs
 
-### Inputs
+* The domain name to be tested ("Child Zone").
 
-The domain name to be tested.
+## Ordered description of steps to be taken to execute the test case
 
-### Ordered description of steps to be taken to execute the test case
+1. Create a DNSKEY query with DO flag set for the apex of the 
+   *Child Zone*.
 
-1. Obtain a set of name server IP addresses using [Method4] and [Method5].
-2. Create a DNSKEY query with DO flag set for the apex of the the zone.
-3. Send the DNSKEY query over UDP to each name server IP address until
-   a response is received or until the set is exhausted.
-4. The DNSKEY algorithm numbers are derived from all the DNSKEY RRs.
-5. The algorithm numbers are matched with the IANA DNSKEY algorithm table.
-6. If any the algorithm numbers is unassigned, reserved, a private algorithm
-   or deprecated, this test case fails.
+2. Retrieve all name server IP addresses for the
+   *Child Zone* using [Method4] and [Method5].
 
-### Outcome(s)
+3. Repeat the following steps for each name server IP address:
 
-If any DNSKEY algorithm number is unassigned, reserved, a private algorithm
-or deprecated, this test case fails.
+   1. Send the DNSKEY query over UDP.
+   2. If no DNS response is returned, then emit *[NO_RESPONSE]*.
+   3. If the DNS response does not contain an DNSKEY RRset,
+      then emit *[NO_RESPONSE_DNSKEY]*.
+   4. Extract the algorithm numbers from each DNSKEY record and
+      compare the extracted algorithm number to the IANA
+      [DNSSEC Algorithm Numbers] registry.
+      1. If the algorithm is classified as "deprecated" (algorithm 
+         1), emit *[ALGORITHM_DEPRECATED]*.
+      2. If the algorithm is classified as "reserved" (algorithm 
+         4, 9, 11, 123-251 or 255), emit *[ALGORITHM_RESERVED]*.
+      3. If the algorithm is classified as "unassigned" (algorithm
+         17-122), emit *[ALGORITHM_UNASSIGNED]*.
+      4. If the algorithm is classified as "private algorithm"
+         (algorithm 253 or 254), emit *[ALGORITHM_PRIVATE]*.
+      5. If the algorithm is classified as "delete DS" (algorithm
+         0), emit *[ALGORITHM_DELETE_DS]*.
+      6. If the algorithm is classified as "indirect key" (algoritm
+         252), emit *[ALGORITHM_INDIRECT_KEY]*.
+      7. If the algorithm is not meant for zone signing (algorithm
+         other than 3, 5-8, 10, 12-16, 253 and 254), emit 
+         *[ALGORITHM_NOT_ZONE_SIGN]*.
+      8. If no message has been emitted for the DNSKEY, emit 
+         *[ALGORITHM_OK]*.
```

Comment:
Since all other steps list explicit algo numbers, this one might as well.

Reply:
Do you ask for redundancy?

Reply:
Fixed.